Track path calculation and momentum at AHDC exit in AHDC Kalman filter

efuchey opened the pull request:
Added in AHDC Kalman filter:
* track total path length;
* track average dEdx (path length divided by ADC sum)
* track momentum exiting the AHDC (improperly called "p_drift");

efuchey committed:
Added track path, momentum drift, and dE/dx calculations in ALERT AHDC Kalman Filter.

efuchey committed:
In AHDC kalman filter calculated the variables:
 "kf_path" (total track path from target to exit);
 "p_drift" (misnomer: momentum of the particle existing the AHDC);
 "dEdx" (ADC sum divided by total path);

efuchey committed:
Fix to the track path calculation:
new kfitter set up with final track state to calculate track path from vertex to exit
instead of existing kfitter to clean residual offest from the fit calculation.
This new kfitter is also used to caclulate post-fit residuals.

efuchey committed:
removed an overlooked printout.

baltzell commented:
Can you make the names "kfitter" and "kFitter" more informative?  e.g. kfPre/kfPost/kfInit/...

efuchey committed:
Renamed KFitter instances:
"kFitter" which fits the track has been renamed into "TrackFitter";
"kfitter" which propagates the track after the fit to calculate the path, momentum at exit, and residuals, has been renamed "PostFitPropagator".
Changes are purely for clarity and do not change the output (tested).

efuchey commented:
"kFitter" which fits the track has been renamed into "TrackFitter";
"kfitter" which propagates the track after the fit to calculate the path, momentum at exit, and residuals, has been renamed "PostFitPropagator".
Changes are purely for clarity and do not change the output (tested).
